specs/networking/messaging.md

```diff
@@ -0,0 +1,41 @@
+ETH 2.0 Networking Spec - Messaging
+===
+
+# Abstract
+
+This specification describes how individual Ethereum 2.0 messages are represented on the wire.
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL”, NOT", “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+# Motivation
+
+This specification seeks to define a messaging protocol that is flexible enough to be changed easily as the ETH 2.0 specification evolves.
+
+# Specification
+
+## Message Structure
+
+An ETH 2.0 message consists of a single byte representing the message version followed by the encoded, potentially compressed body. We separate the message's version from the version included in the `libp2p` protocol path in order to allow encoding and compression schemes to be updated independently of the `libp2p` protocols themselves.
```

Comment:
This is similar to what Raul K. from IPFS was saying - I can update the spec to reflect that exchange. What do you think about upgradeability paths if we go this route? From what I understand, there's a possibility that hybrid encoding formats will be supported and from an implementation perspective it might be easier to simply read the first byte of the message to determine how messages will be serialized.

Reply:
- Should the encoded message also be length-prefixed? 
SSZ doesn't do this out of the box, and this will allow implementations to know incoming message sizes and therefore how much to read or reject excessive sized messages (potential DOS). 
- Should there be a size limit to messages?

Reply:
`libp2p` length-prefixes the message automatically.

I do think there should be a size limit.

Reply:
I'm not sure about go-libp2p, but I've implemented a basic implementation of this in rust-libp2p. I had to specify length prefixes and size limits (to know when to stop reading from the stream). This may be the case for other implementations, I didn't see length prefixes in the libp2p specs.

Reply:
It's defined in here, I believe: https://github.com/multiformats/multistream-select.